internal/s3storage/s3storage.go

```diff
@@ -0,0 +1,84 @@
+package s3storage
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/lunarway/release-manager/internal/artifact"
+	"github.com/lunarway/release-manager/internal/log"
+)
+
+type Service struct {
+	bucketName string
+	logger     *log.Logger
+	s3client   *s3.S3
+}
+
+func Initialize(logger *log.Logger) (*Service, error) {
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String("eu-west-1"),
```

Comment:
Would be nice if this was configurable along with the bucket name. But we can always add that in another PR. :)

Reply:
Yeah, like the auth options as arguments too 😊 let's grab them all together